### PR TITLE
Fix readList deprecation

### PR DESCRIPTION
**Pull request by RobozinhoD**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
part of #12364

## How Has This Been Tested?

Opened the help menu on 21 and 33 emulators

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/RecursivePictureMenu.kt [hunk 1]
```diff
@@ -139,11 +140,10 @@ class RecursivePictureMenu : DialogFragment() {
             }
         }
 
-        @Suppress("deprecation") // readList
         protected constructor(parcel: Parcel) : super(parcel) {
             if (parcel.readByte().toInt() == 0x01) {
                 mChildren = ArrayList()
-                parcel.readList(mChildren, Item::class.java.classLoader)
+                parcel.readListCompat(mChildren, Item::class.java.classLoader, Item::class.java)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/RecursivePictureMenu.kt:146]: Can we make this `parcel.readListCompat<Item>(mChildren)` using `reified`

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by david-allison** [comment]
conflicts

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/RecursivePictureMenu.kt (hunk 1 above)
Comment (RobozinhoD) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/RecursivePictureMenu.kt]: done

**Review by david-allison — APPROVED** [review]
Much nicer

**Review by mikehardy — APPROVED** [review]
Looks great thanks
maintainer note: checked lint locally after update to main since ktlint changes were in flight in parallel, ✅

**Pull request merged**